dhall-docs/src/Dhall/Docs.hs

```diff
@@ -51,20 +72,157 @@ parseOptions :: Parser Options
 parseOptions =
         Options
     <$> Options.Applicative.strOption
-        ( Options.Applicative.long "path"
-       <> Options.Applicative.metavar "PATH"
-       <> Options.Applicative.help "Root folder of your dhall package" )
+        ( Options.Applicative.long "input"
+       <> Options.Applicative.metavar "INPUT"
+       <> Options.Applicative.help "Directory of your dhall package" )
     <*> parseStrategy
+    <*> Options.Applicative.strOption
+        ( Options.Applicative.long "output"
+       <> Options.Applicative.metavar "OUTPUT"
+       <> Options.Applicative.help "Directory where your docs will be generated"
+       <> Options.Applicative.value "docs" )
 
 -- | `ParserInfo` for the `Options` type
 parserInfoOptions :: ParserInfo Options
 parserInfoOptions =
+    let progDesc = "Generate HTML documentation from a dhall package or file" in
     Options.Applicative.info
         (Options.Applicative.helper <*> parseOptions)
         (   Options.Applicative.fullDesc
-        <>  Options.Applicative.progDesc "Generate HTML documentation from a dhall package or file"
+        <>  Options.Applicative.progDesc progDesc
         )
 
+{-| Fetches a list of all dhall files in a directory. This is not the same
+    as finding all files that ends in @.dhall@, but finds all files that
+    successfully parses as a valid dhall file.
+
+    The reason it doesn't guide the search by its extension is because of the
+    dhall <https://prelude.dhall-lang.org Prelude>
+    That package doesn't ends any of their files in @.dhall@.
+-}
+getAllDhallFiles
+    :: Path Abs Dir -- ^ Base directory to do the search
+    -> IO [(Path Abs File, Header)]
+getAllDhallFiles baseDir = do
+    files <- snd <$> Path.IO.listDirRecur baseDir
+    Data.Maybe.catMaybes <$> mapM readDhall files
+  where
+    readDhall :: Path Abs File -> IO (Maybe (Path Abs File, Header))
+    readDhall absFile = do
+        let filePath = Path.fromAbsFile absFile
+        fileContents <- Data.ByteString.readFile filePath
+        return $ case Text.Encoding.decodeUtf8' fileContents of
+            Left _ -> Nothing
+            Right contents ->
+                case exprAndHeaderFromText filePath contents of
+                    Right (header, _) -> return (absFile, header)
+                    _ -> Nothing
+
+
+{-| Calculate the relative path needed to access files on the first argument
+    relative from the second argument.
+
+    The second argument needs to be a child of the first, otherwise it will
+    loop forever
+
+    Examples:
+
+    >>> resolveRelativePath $(mkAbsDir "/a/b/c/") $(mkAbsDir "/a/b/c/d/e/")
```

Comment:
Personally, I think the QuasiQuoters look a bit nicer. It doesn't really matter though. ;)

```suggestion
    >>> resolveRelativePath [absdir|/a/b/c/] [absdir|/a/b/c/d/e/]
```